packages/core/src/Stores/Modules/P2P/p2p-store.js

```diff
@@ -0,0 +1,72 @@
+import { action, observable } from 'mobx';
+import routes from '@deriv/shared/utils/routes';
+import { SendbirdAPI, P2pStorage } from '@deriv/p2p/lib/utils';
+import { WS } from 'Services';
+import BaseStore from '../../base-store';
+
+export default class P2pStore extends BaseStore {
+    constructor({ root_store }) {
+        super({ root_store });
+        this.onClientInit(this.initSendbird);
+        this.onLogout(this.disconnectSendbird);
+    }
+
+    @observable unread_notification_count = 0;
+    @observable is_advertiser = false;
+    @observable is_visible = false;
+
+    @action.bound
+    async initSendbird() {
+        const { client } = this.root_store;
+
+        if (client.is_logged_in) {
+            P2pStorage.setLoginId(client.loginid);
+            P2pStorage.addNotificationListener(this.setUnreadNotificationCount);
+
+            const p2p_advertiser_info = await WS.authorized.p2pAdvertiserInfo();
+            const p2p_service_token = await WS.authorized.serviceToken({ service_token: 1, service: 'sendbird' });
+
+            const sendbird_user_id = p2p_advertiser_info?.p2p_advertiser_info?.chat_user_id;
+            const service_token = p2p_service_token?.service_token?.sendbird?.token;
+
+            if (!sendbird_user_id || !service_token) {
+                // eslint-disable-next-line no-console
+                console.warn('Could not initialise Sendbird API for notifications');
+                return Promise.resolve();
+            }
+
+            this.sendbird_api = new SendbirdAPI({ LocalStorage: P2pStorage });
+            this.sendbird_api.init(sendbird_user_id, service_token);
+        }
+
+        return Promise.resolve();
+    }
+
+    @action.bound
+    async disconnectSendbird() {
+        if (this.sendbird_api) {
+            this.sendbird_api.disconnect();
+        }
+
+        return Promise.resolve();
```

Comment:
[Nit] You don't need to make the function `async` and `return Promise.resolve()`, just one or the other will do.

Reply:
Heh, true! 👍